kratos/utilities/sub_model_part_operations_utility.h

```diff
@@ -0,0 +1,167 @@
+ * @author Miguel Maso Sotomayor
+ * @ingroup KratosCore
+*/
+class KRATOS_API(KRATOS_CORE) SubModelPartOperationsUtility
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of SubModelPartOperationsUtility
+    KRATOS_CLASS_POINTER_DEFINITION(SubModelPartOperationsUtility);
+
+    typedef std::size_t IndexType;
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    enum class SetOperators {Union, Intersection, Difference};
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    /// Destructor.
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TEntityType, class TContainerType>
+    void Union(ModelPart& rModelPartA, ModelPart& rModelPartB, ModelPart& rDestination)
+    {
+        SetOperation<TEntityType, TContainerType>(
+            rModelPartA, rModelPartB, rDestination, SetOperators::Union);
+    }
+
+    template<class TEntityType, class TContainerType>
+    void Intersection(ModelPart& rModelPartA, ModelPart& rModelPartB, ModelPart& rDestination)
+    {
+        SetOperation<TEntityType, TContainerType>(
+            rModelPartA, rModelPartB, rDestination, SetOperators::Intersection);
+    }
+
+    template<class TEntityType, class TContainerType>
+    void Difference(ModelPart& rModelPartA, ModelPart& rModelPartB, ModelPart& rDestination)
+    {
+        SetOperation<TEntityType, TContainerType>(
+            rModelPartA, rModelPartB, rDestination, SetOperators::Difference);
+    }
+
+    ///@}
+
+private:
+    ///@name Private Operations
+    ///@{
+
+    template<class TEntityType>
+    void AddEntities(const std::vector<IndexType>& rIds, ModelPart& rModelPart);
+
+    template<class TContainerType>
+    TContainerType& GetContainer(ModelPart& rModelPart);
+
+    template<class TContainerType>
+    std::vector<IndexType> GetContainerIds(ModelPart& rModelPart)
+    {
+        TContainerType& r_container = GetContainer<TContainerType>(rModelPart);
+        std::vector<IndexType> ids_vector;
+        for (auto it = r_container.begin(); it != r_container.end(); ++it)
+        {
+            ids_vector.push_back(it->Id());
+        }
+        return ids_vector;
+    }
+
+    template<class TEntityType, class TContainerType>
+    void SetOperation(
+        ModelPart& rModelPartA,
+        ModelPart& rModelPartB,
+        ModelPart& rDestination,
+        SetOperators ThisOperator)
+    {
+        std::vector<IndexType> ids_a = GetContainerIds<TContainerType>(rModelPartA);
+        std::vector<IndexType> ids_b = GetContainerIds<TContainerType>(rModelPartB);
+        std::vector<IndexType> ids_destination;
+        std::sort(ids_a.begin(), ids_a.end());
+        std::sort(ids_b.begin(), ids_b.end());
+
+        if (ThisOperator == SetOperators::Union) {
+            std::set_union(
+                ids_a.begin(), ids_a.end(),
+                ids_b.begin(), ids_b.end(),
+                std::back_inserter(ids_destination));
+        }
```

Comment:
```suggestion 
        } else if (ThisOperator == SetOperators::Intersection) {
```
From my side everything is OK, except the style of these if...

Reply:
nice to see you @loumalouomega! I find the extra line more readable, however, I've pushed the suggestions